### PR TITLE
[FLINK-5209] [webfrontend] Fix TaskManager metrics

### PR DESCRIPTION
**Pull request by greghogan**

### CODE REVIEW

File: flink-runtime-web/web-dashboard/app/partials/taskmanager/taskmanager.metrics.jade [hunk 1]
```diff
@@ -75,18 +75,18 @@ div(ng-if="metrics.id")
...
-        td {{ metrics.metrics.directTotal }}
+        td {{ metrics.metrics.directCount | humanizeBytes }}
+        td {{ metrics.metrics.directUsed | humanizeBytes }}
+        td {{ metrics.metrics.directMax | humanizeBytes }}
       tr
         td Mapped
-        td {{ metrics.metrics.mappedCount }}
-        td {{ metrics.metrics.mappedUsed }}
-        td {{ metrics.metrics.mappedMax }}
+        td {{ metrics.metrics.mappedCount | humanizeBytes }}
```
Comment (zentol) [inline flink-runtime-web/web-dashboard/app/partials/taskmanager/taskmanager.metrics.jade]: The unit for none of these metrics is bytes

File: flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/TaskManagersHandler.java [hunk 2]
```diff
@@ -124,13 +124,21 @@ public String handleJsonRequest(Map<String, String> pathParams, Map<String, Stri
 							gen.writeNumberField("totalUsed", heapUsed + nonHeapUsed);
 							gen.writeNumberField("totalMax", heapTotal + nonHeapTotal);
 
-							gen.writeStringField("directCount", metrics.getMetric("Status.JVM.Memory.Direct.Count", "0"));
-							gen.writeStringField("directUsed", metrics.getMetric("Status.JVM.Memory.Direct.MemoryUsed", "0"));
-							gen.writeStringField("directMax", metrics.getMetric("Status.JVM.Memory.Direct.TotalCapacity", "0"));
+							long directCount = Long.valueOf(metrics.getMetric("Status.JVM.Memory.direct.Count", "0"));
```
Comment (zentol) [inline flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/TaskManagersHandler.java]: is there a benefit to passing them as longs?

File: flink-runtime-web/web-dashboard/app/partials/taskmanager/taskmanager.metrics.jade (hunk 1 above)
Comment (zentol) [inline flink-runtime-web/web-dashboard/app/partials/taskmanager/taskmanager.metrics.jade]: wait; only mapped/direct-count are not measured in bytes.

File: flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/TaskManagersHandler.java [hunk 3]
```diff
@@ -124,13 +124,21 @@ public String handleJsonRequest(Map<String, String> pathParams, Map<String, Stri
...
+
+							long mappedCount = Long.valueOf(metrics.getMetric("Status.JVM.Memory.mapped.Count", "0"));
+							long mappedUsed = Long.valueOf(metrics.getMetric("Status.JVM.Memory.mapped.MemoryUsed", "0"));
+							long mappedMax = Long.valueOf(metrics.getMetric("Status.JVM.Memory.mapped.TotalCapacity", "0"));
+
+							gen.writeNumberField("mappedCount", mappedCount);
+							gen.writeNumberField("mappedUsed", mappedUsed);
+							gen.writeNumberField("mappedMax", mappedMax);
 
 							gen.writeStringField("memorySegmentsAvailable", metrics.getMetric("Status.Network.AvailableMemorySegments", "0"));
```
Comment (zentol) [inline flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/TaskManagersHandler.java]: If you change the above to write longs these can be changed as well, along with with the GC metrics.

File: flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/TaskManagersHandler.java (hunk 2 above)
Comment (greghogan) [inline flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/TaskManagersHandler.java]: Some of the numeric metrics were passed as long and some as string. I would expect someone consuming the JSON to prefer numeric values.

File: flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/TaskManagersHandler.java (hunk 3 above)
Comment (greghogan) [inline flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/TaskManagersHandler.java]: Will do ...

File: flink-runtime-web/web-dashboard/app/partials/taskmanager/taskmanager.metrics.jade (hunk 1 above)
Comment (greghogan) [inline flink-runtime-web/web-dashboard/app/partials/taskmanager/taskmanager.metrics.jade]: I'll add a Coffee filter to localize the numbers with commas / periods.

**Comment by zentol** [comment]
Looks good to merge, +1 :)

**Pull request closed without merging**